src/org/jitsi/impl/neomedia/rtcp/RTCPIterator.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright @ 2015 Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.impl.neomedia.rtcp;
+
+import net.sf.fmj.media.rtp.*;
+import org.jitsi.impl.neomedia.*;
+import org.jitsi.service.neomedia.*;
+
+import java.util.*;
+
+/**
+ * An {@code Iterator} for RTCP packets contained in a compound RTCP packet.
+ * For a {@code PacketTransformer} that splits compound RTCP packets into
+ * individual RTCP packets {@see CompoundPacketEngine}.
+ *
+ * Instances of this class are not thread-safe. If multiple threads access an
+ * instance concurrently, it must be synchronized externally.
+ *
+ * @author George Politis
+ */
+public class RTCPIterator
+    implements Iterator<ByteArrayBuffer>
+{
+    /**
+     * The {@code RawPacket} that holds the RTCP packet to iterate.
+     */
+    private final ByteArrayBuffer baf;
+
+    /**
+     * The offset in the {@link #baf} where the next packet is to be looked for.
+     */
+    private int nextOff;
+
+    /**
+     * The remaining length in {@link #baf}.
+     */
+    private int remainingLen;
+
+    /**
+     * The length of the last next element.
+     */
+    private int lastLen;
+
+    /**
+     * Ctor.
+     *
+     * @param baf The {@code ByteArrayBuffer} that holds the compound RTCP
+     * packet to iterate.
+     */
+    public RTCPIterator(ByteArrayBuffer baf)
+    {
+        this.baf = baf;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean hasNext()
+    {
+        return RTCPHeaderUtils.getLength
+            (baf.getBuffer(), nextOff, remainingLen) >= RTCPHeader.SIZE;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public ByteArrayBuffer next()
+    {
+        int pktLen = RTCPHeaderUtils.getLength(
+            baf.getBuffer(), nextOff, remainingLen);
+        if (pktLen < RTCPHeader.SIZE)
+        {
+            throw new IllegalStateException();
+        }
+
+        RawPacket next = new RawPacket(baf.getBuffer(), nextOff, pktLen);
```

Comment:
Here we use nextOff, but it was only initialized to 0. I think it should be initialized to baf.getOffset() when we create the iterator.